lib/main.dart

```diff
@@ -13,103 +14,319 @@ class MyApp extends StatelessWidget {
+              Text(
+                'rain w/ t1de (s&r)',
+                style: TextStyle(
+                    fontSize: 15,
+                    color: Colors.white,
+                    fontWeight: FontWeight.w500),
+              ),
+              Text(
+                'sindye',
+                style: TextStyle(
+                    fontSize: 13,
+                    color: Colors.grey,
+                    fontWeight: FontWeight.w100),
+              ),
+            ],
+          ),
+          Container(width: width2),
+          IconButton(
+              onPressed: () {},
+              icon: const Icon(Icons.favorite, color: Colors.white)),
+        ],
+      )));
+}
+
+Widget _relatedSongsList(BuildContext context) {
+  String title = 'More of what you like';
+  String description = 'Suggestions based on what you\'ve liked or played';
+  var songsImages = [
+    'assets/1_1.jpg',
+    'assets/1_2.jpg',
+    'assets/1_3.jpg',
+    'assets/1_4.jpg'
+  ];
+  var songsTitles = [
+    'Your Pink Hair (...',
+    'Digital Love (Fea...',
+    'reach you + tide...',
+    'Sony Ericsson'
+  ];
+  String songTheme = 'Related tracks';
+  return _songsList(
+      context, title, description, songsImages, songsTitles, songTheme);
+}
+
+Widget _chartSongsList(BuildContext context) {
+  String title = 'Charts: Top 50';
+  String description = 'The most played tracks on SoundCloud this week';
+  var songsImages = [
+    'assets/2_1.jpg',
+    'assets/2_1.jpg',
+    'assets/2_2.jpg',
+    'assets/2_3.jpg'
+  ];
+  var songsTitles = [
+    'All music genres',
+    'Global Beats',
+    'Hip Hop & Rap',
+    'Electronic'
+  ];
+  String songTheme = 'Top 50';
+  return _songsList(
+      context, title, description, songsImages, songsTitles, songTheme);
+}
+
+Widget _songsList(BuildContext context, String title, String description,
+    songsImages, songsTitles, String songTheme) {
+  return Container(
+    margin: const EdgeInsets.only(top: 20),
+    child: Column(children: [
+      Container(
+          margin: const EdgeInsets.only(left: 20),
+          child: Align(
+              alignment: Alignment.topLeft,
+              child: Text(
+                title,
+                style: const TextStyle(
+                    fontSize: 25,
+                    color: Colors.white,
+                    fontWeight: FontWeight.w100),
+              ))),
+      Container(
+          margin: const EdgeInsets.only(left: 20, top: 5),
+          child: Align(
+              alignment: Alignment.topLeft,
+              child: Text(
+                description,
+                style: const TextStyle(
+                    fontSize: 15,
+                    color: Colors.grey,
+                    fontWeight: FontWeight.w100),
+              ))),
+      SingleChildScrollView(
+        scrollDirection: Axis.horizontal,
+        child: Row(children: <Widget>[
+          _songContainer(context, songsImages[0], songsTitles[0], songTheme),
```

Comment:
Pretty nice code, perfect app, and musical taste, but  I recommended using a map (**songs.map((song) =>_songContainer(context, song.img, song.title, songTheme)).toList()**)